### PR TITLE
Add instrumentation tests for storage plugin (SAF)

### PR DESCRIPTION
**Pull request by grote**
First part of  #95

### CODE REVIEW

File: app/src/main/java/com/stevesoltys/seedvault/settings/SettingsManager.kt [hunk 1]
```diff
@@ -49,6 +49,7 @@ class SettingsManager(context: Context) {
         return Storage(uri, name, isUsb)
     }
 
+    // TODO find a better solution for this hack abusing the settings manager
```
Comment (stevesoltys) [inline app/src/main/java/com/stevesoltys/seedvault/settings/SettingsManager.kt:52]: Maybe this could be a research task.

File: build.gradle [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 buildscript {
 
-    ext.kotlin_version = '1.3.61'
+    ext.kotlin_version = '1.3.72'
```
Comment (chirayudesai) [inline build.gradle:5]: Need to test this with #98

Comment (grote) [reply build.gradle:5]: Might be good to know what Kotlin version AOSP10 uses so we can stay in sync with that.

**Comment by chirayudesai** [comment]
Is "Upgrade gradle and kotlin" needed for this PR? If not, could merge without that, or just the gradle update.

File: build.gradle (hunk 2 above)
Comment (chirayudesai) [reply build.gradle:5]: 1.3.21 in the latest Android 10 release (august)
1.3.72 in master, let's see what R / 11 has.

Source: https://android.googlesource.com/platform/external/kotlinc/+/refs/tags/android-10.0.0_r41/build.txt

Comment (grote) [reply build.gradle:5]: Hmm... so maybe it is better to downgrade to that? I *think* that all 1.3.x should be compatible, but who knows...

**Comment by grote** [comment]
> Is "Upgrade gradle and kotlin" needed for this PR? If not, could merge without that, or just the gradle update.

The gradle update shouldn't be an issue as it isn't used for AOSP builds. But we should decide what we do with the kotlin compiler version.

File: build.gradle (hunk 2 above)
Comment (chirayudesai) [reply build.gradle:5]: Not sure. So far it hasn't been an issue, but yeah might be better to stay in sync.

But then again 11 isn't too far out, so let's decide then?

**Comment by chirayudesai** [comment]
> > Is "Upgrade gradle and kotlin" needed for this PR? If not, could merge without that, or just the gradle update.
> 
> The gradle update shouldn't be an issue as it isn't used for AOSP builds. But we should decide what we do with the kotlin compiler version.

Yep, so could you drop the kotlin bit from this commit/PR for now, we can deal with that later.

File: build.gradle (hunk 2 above)
Comment (grote) [reply build.gradle]: Ok so I reverted this change and left the Kotlin version as it was.

**Comment by chirayudesai** [comment]
Thanks!

**Pull request merged**